src/frontend/screens/Achievements/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react'
+import walletState from 'frontend/state/WalletState'
+import { AchievementCard, AchievementSummaryTable } from '@hyperplay/ui'
+import { NavLink } from 'react-router-dom'
+import { StatusIconState } from '@hyperplay/ui/dist/components/AchievementCard/components/StatusIcon'
+import { useTranslation } from 'react-i18next'
+import { observer } from 'mobx-react-lite'
+import AchievementState, {
+  ACHIEVEMENT_SORT_OPTIONS
+} from 'frontend/state/AchievementState'
+import MintAchievementsState from 'frontend/state/MintAchievementsState'
+
+export default observer(function Achievements(): JSX.Element {
+  const { t } = useTranslation()
+
+  const numFreeMints = AchievementState.numFreeMints
+
+  const achievementsToBeMinted = MintAchievementsState.achievementsToBeMinted
+  const toggleAchievementToBeMinted =
+    MintAchievementsState.toggleAchievementToBeMinted
+  const isLoading = MintAchievementsState.isLoading
+  const handleMint = MintAchievementsState.handleMint
+  const handleUpdate = MintAchievementsState.handleUpdate
+  const achievementsToBeUpdated = MintAchievementsState.achievementsToBeUpdated
+  const toggleAchievementToBeUpdated =
+    MintAchievementsState.toggleAchievementToBeUpdated
+
+  const isDisabled = isLoading || !walletState.isConnected
+
+  if (AchievementState.summaryAchievementsToDisplay === undefined) {
+    AchievementState.fetchMoreSummaryAchievements()
+  }
+
+  const tabs = [
+    { value: 'all', label: 'All' },
+    { value: 'new', label: 'New' }
+  ]
+  if (AchievementState.mintedAchievements > 0) {
+    tabs.push({ value: 'minted', label: 'Minted' })
+  }
+
+  const imagesToPreload =
+    AchievementState?.summaryAchievementsToDisplay?.map(
+      (game) => game.gameImageURL
+    ) ?? []
+
+  return (
+    <>
+      <AchievementSummaryTable
+        imagesToPreload={imagesToPreload}
+        games={
+          AchievementState.summaryAchievementsToDisplay
+            ? AchievementState.summaryAchievementsToDisplay.map((game) => {
+                const id = String(game.gameId)
+                const isUpdate =
+                  game.isNewAchievement && game.mintedAchievementCount > 0
+                const state = !walletState.isConnected
```

Comment:
I know this was not added by you but in this case I think having more branches is more readable:

```ts
let state;

if (!walletState.isConnected) {
  state = 'disabled';
} else if (
  achievementsToBeMinted.includes(id) ||
  achievementsToBeUpdated.includes(id)
) {
  state = 'active';
} else if (isUpdate) {
  state = 'update';
} else {
  state = 'default';
}
```

Reply:
awesome. thx for the snippet. just updated